packages/aggregate-client/src/aggregate.js

```diff
@@ -0,0 +1,97 @@
+import * as AggregateCapabilities from '@web3-storage/capabilities/aggregate'
+import { CBOR } from '@ucanto/core'
+
+import { servicePrincipal, connection } from './service.js'
+
+export const MIN_SIZE = 1 + 127 * (1 << 27)
+export const MAX_SIZE = 127 * (1 << 28)
+
+/**
+ * Offer an aggregate to be assembled and stored.
+ *
+ * @param {import('./types').InvocationConfig} conf - Configuration
+ * @param {import('./types').Offer[]} offers
+ * @param {import('./types').RequestOptions} [options]
+ */
+export async function aggregateOffer(
+  { issuer, with: resource, proofs, audience },
+  offers,
+  options = {}
+) {
+  /* c8 ignore next */
+  const conn = options.connection ?? connection
+
+  // TODO: Get commitmentProof
+  const commitmentProof = 'commitmentProof'
+
+  // Validate size for offer is valid
+  const size = offers.reduce((accum, offer) => accum + offer.size, 0)
+  if (size < MIN_SIZE) {
+    throw new Error(
+      `provided size is not enough to create an offer (${size} < ${MIN_SIZE})`
+    )
+  } else if (size > MAX_SIZE) {
+    throw new Error(
+      `provided size is larger than it can be accepted for an offer (${size} > ${MAX_SIZE})`
+    )
+  }
```

Comment:
I'd remove this and let the server validate - different implementations might allow different min/max sizes and want to use the same client.